lib/pkcs7.js

```diff
@@ -342,16 +342,20 @@ p7.createSignedData = function() {
         if('content' in msg) {
           var content;
           if(msg.content instanceof forge.util.ByteBuffer) {
-            content = msg.content.bytes();
+            content = asn1.create(asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, false,
+              msg.content.bytes());
           } else if(typeof msg.content === 'string') {
-            content = forge.util.encodeUtf8(msg.content);
+            content = asn1.create(asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, false,
+              forge.util.encodeUtf8(msg.content));
+          } else if (typeof msg.content === 'object') {
+            content = asn1.create(asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, true,
+              [msg.content]);
```

Comment:
So, looking back at this code, this section of code only executes as a helper to automatically generate `contentInfo` in a common case. I think there was a presumption that the `contentInfo` would be generated externally unless `msg.content` is set to a byte buffer or a string. However, the conditional seems to allow this code to execute if `msg.content` is an `object` ... and I'm not sure why yet. I'll need to see if there is well defined behavior in this case in the documentation, but it seems to me that `msg.content` as an `object` could imply a number of different things and we can't simply pick to encode it like this.

In other words, if your `msg.content` is a string or a byte buffer, we have a well defined way in which it will be automatically wrapper. If you need to do something else, you should generate your own `contentInfo`. If a simple tweaking to this code better handles some other common cases that's fine, but we need to say explicitly what's going to happen if you pass in `msg.content` as an object -- and it's not yet clear to me that what's going on here is the "right" thing (or the highest utility thing) to do.